commit 3d96840cd9cb24d2f3d5a76b6623e5da7bdc5154
Author: popcornmix <popcornmix@gmail.com>
Date:   Tue Aug 19 00:25:00 2014 +0100

    bcm2708: Enable bcm2835_mmc by default

diff --git a/arch/arm/mach-bcm2708/bcm2708.c b/arch/arm/mach-bcm2708/bcm2708.c
index cf421ee..ac30e14 100644
--- a/arch/arm/mach-bcm2708/bcm2708.c
+++ b/arch/arm/mach-bcm2708/bcm2708.c
@@ -92,7 +92,7 @@ static unsigned disk_led_active_low = 1;
 static unsigned reboot_part = 0;
 static unsigned w1_gpio_pin = W1_GPIO;
 static unsigned w1_gpio_pullup = W1_PULLUP;
-static unsigned bcm2835_mmc = 0;
+static unsigned bcm2835_mmc = 1;
 
 static void __init bcm2708_init_led(void);
 
