commit fb55443b3e9f97ebd4775776ab9b1c00bd1134ea
Author: Sven Köhler <sven.koehler@gmail.com>
Date:   Mon Aug 7 18:49:20 2017 +0200

    Fix dependencies broken since driver was renamed

diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index e577d20963bc..324da890a83a 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -25,7 +25,7 @@ config MMC_BCM2835_DMA
 
 config MMC_BCM2835_PIO_DMA_BARRIER
 	int "Block count limit for PIO transfers"
-	depends on MMC_BCM2835 && MMC_BCM2835_DMA
+	depends on MMC_BCM2835_MMC && MMC_BCM2835_DMA
 	range 0 256
 	default 2
 	help
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1185_fb55443b3e9f97ebd4775776ab9b1c00bd1134ea.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1185_fb55443b3e9f97ebd4775776ab9b1c00bd1134ea.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
