commit 854f2b38f14d66c326fa897ad906d640239251c6
Author: Phil Elwell <pelwell@users.noreply.github.com>
Date:   Tue Jan 3 21:27:46 2017 +0000

    Document the si7020 option

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 81d991803be3..e8fa4ccb44c3 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -549,6 +549,9 @@ Params: bmp085                  Select the Bosch sensortronic BMP085
 
         bmp280                  Select the Bosch sensortronic BMP280
 
+        si7020                  Select the Silicon Labs Si7013/20/21 humidity/
+                                temperature sensor
+
 
 Name:   i2c0-bcm2708
 Info:   Enable the i2c_bcm2708 driver for the i2c0 bus. Not all pin combinations
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1115_854f2b38f14d66c326fa897ad906d640239251c6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1115_854f2b38f14d66c326fa897ad906d640239251c6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
