commit 3583cfac023b4e680f4e625a5027180c478ee15c
Author: Serge Schneider <serge@raspberrypi.org>
Date:   Tue Jan 29 12:05:49 2019 +0000

    mfd: Add rpi_sense_core of compatible string

diff --git a/drivers/mfd/rpisense-core.c b/drivers/mfd/rpisense-core.c
index eea9312dc96a..6cfd63e5e8b8 100644
--- a/drivers/mfd/rpisense-core.c
+++ b/drivers/mfd/rpisense-core.c
@@ -138,6 +138,14 @@ static const struct i2c_device_id rpisense_i2c_id[] = {
 };
 MODULE_DEVICE_TABLE(i2c, rpisense_i2c_id);
 
+#ifdef CONFIG_OF
+static const struct of_device_id rpisense_core_id[] = {
+	{ .compatible = "rpi,rpi-sense" },
+	{ },
+};
+MODULE_DEVICE_TABLE(of, rpisense_core_id);
+#endif
+
 
 static struct i2c_driver rpisense_driver = {
 	.driver = {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1474_3583cfac023b4e680f4e625a5027180c478ee15c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1474_3583cfac023b4e680f4e625a5027180c478ee15c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
