commit 64508dd6e5cd172898b2660e2de9c33d8b3d9e26
Author: popcornmix <popcornmix@gmail.com>
Date:   Mon Jul 25 17:06:50 2016 +0100

    iqaudio-dac: Compile fix - untested

diff --git a/sound/soc/bcm/iqaudio-dac.c b/sound/soc/bcm/iqaudio-dac.c
index 4e8e6dec14bc..aa15bc4b49ca 100644
--- a/sound/soc/bcm/iqaudio-dac.c
+++ b/sound/soc/bcm/iqaudio-dac.c
@@ -76,7 +76,11 @@ static void snd_rpi_iqaudio_gpio_unmute(struct snd_soc_card *card)
 static int snd_rpi_iqaudio_set_bias_level(struct snd_soc_card *card,
 	struct snd_soc_dapm_context *dapm, enum snd_soc_bias_level level)
 {
-	struct snd_soc_dai *codec_dai = card->rtd[0].codec_dai;
+	struct snd_soc_pcm_runtime *rtd;
+	struct snd_soc_dai *codec_dai;
+
+	rtd = snd_soc_get_pcm_runtime(card, card->dai_link[0].name);
+	codec_dai = rtd->codec_dai;
 
 	if (dapm->dev != codec_dai->dev)
 		return 0;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1070_64508dd6e5cd172898b2660e2de9c33d8b3d9e26.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1070_64508dd6e5cd172898b2660e2de9c33d8b3d9e26.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
