commit 97f21471e8e48fecb0b1c1b96fe934d1148dffe2
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Wed Aug 22 07:44:57 2018 +0200

    Linux 4.17.18

diff --git a/Makefile b/Makefile
index 5ff2040cf3ee..429a1fe0b40b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 VERSION = 4
 PATCHLEVEL = 17
-SUBLEVEL = 17
+SUBLEVEL = 18
 EXTRAVERSION =
 NAME = Merciless Moray
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1042_97f21471e8e48fecb0b1c1b96fe934d1148dffe2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1042_97f21471e8e48fecb0b1c1b96fe934d1148dffe2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
