commit d3f40ddff28fa677f0f9261c24a79520ca7ed1cb
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Nov 9 13:02:52 2016 +0000

    bcm: Make RASPBERRYPI_POWER depend on PM

diff --git a/drivers/soc/bcm/Kconfig b/drivers/soc/bcm/Kconfig
index a39b0d58ddd0..e037a6dd79d1 100644
--- a/drivers/soc/bcm/Kconfig
+++ b/drivers/soc/bcm/Kconfig
@@ -4,6 +4,7 @@ config RASPBERRYPI_POWER
 	bool "Raspberry Pi power domain driver"
 	depends on ARCH_BCM2835 || (COMPILE_TEST && OF)
 	depends on RASPBERRYPI_FIRMWARE=y
+	depends on PM
 	select PM_GENERIC_DOMAINS if PM
 	help
 	  This enables support for the RPi power domains which can be enabled
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1022_d3f40ddff28fa677f0f9261c24a79520ca7ed1cb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1022_d3f40ddff28fa677f0f9261c24a79520ca7ed1cb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
