commit be76869a01d585870589fedda3f9bb078fe9cefb
Author: Aaron Shaw <shawaj@gmail.com>
Date:   Thu Dec 28 17:31:52 2017 +0000

    add backlight control to rpi-display overlay

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 6e43bbdd4ef2..e0f29ec6627a 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1329,6 +1329,7 @@ Params: speed                   Display SPI bus speed
         debug                   Debug output level {0-7}
         xohms                   Touchpanel sensitivity (X-plate resistance)
         swapxy                  Swap x and y axis
+        backlight               Change backlight GPIO pin {e.g. 12, 18}
 
 
 Name:   rpi-ft5406
diff --git a/arch/arm/boot/dts/overlays/rpi-display-overlay.dts b/arch/arm/boot/dts/overlays/rpi-display-overlay.dts
index d7e72eef594e..533b5c140b54 100644
--- a/arch/arm/boot/dts/overlays/rpi-display-overlay.dts
+++ b/arch/arm/boot/dts/overlays/rpi-display-overlay.dts
@@ -79,11 +79,13 @@
 		};
 	};
 	__overrides__ {
-		speed =   <&rpidisplay>,"spi-max-frequency:0";
-		rotate =  <&rpidisplay>,"rotate:0";
-		fps =     <&rpidisplay>,"fps:0";
-		debug =   <&rpidisplay>,"debug:0";
-		xohms =   <&rpidisplay_ts>,"ti,x-plate-ohms;0";
-		swapxy =  <&rpidisplay_ts>,"ti,swap-xy?";
+		speed =     <&rpidisplay>,"spi-max-frequency:0";
+		rotate =    <&rpidisplay>,"rotate:0";
+		fps =       <&rpidisplay>,"fps:0";
+		debug =     <&rpidisplay>,"debug:0";
+		xohms =     <&rpidisplay_ts>,"ti,x-plate-ohms;0";
+		swapxy =    <&rpidisplay_ts>,"ti,swap-xy?";
+		backlight = <&rpidisplay>,"led-gpios:4",
+		            <&rpi_display_pins>,"brcm,pins:0";
 	};
 };
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1160_be76869a01d585870589fedda3f9bb078fe9cefb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1160_be76869a01d585870589fedda3f9bb078fe9cefb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
