commit 0e5399322277cbfec7b794b1afacfc2c6ab43e57
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Nov 9 13:02:52 2016 +0000

    bcm: Make RASPBERRYPI_POWER depend on PM

diff --git a/drivers/soc/bcm/Kconfig b/drivers/soc/bcm/Kconfig
index a39b0d58ddd0..e037a6dd79d1 100644
--- a/drivers/soc/bcm/Kconfig
+++ b/drivers/soc/bcm/Kconfig
@@ -4,6 +4,7 @@ config RASPBERRYPI_POWER
 	bool "Raspberry Pi power domain driver"
 	depends on ARCH_BCM2835 || (COMPILE_TEST && OF)
 	depends on RASPBERRYPI_FIRMWARE=y
+	depends on PM
 	select PM_GENERIC_DOMAINS if PM
 	help
 	  This enables support for the RPi power domains which can be enabled
