commit 4a62ccc0e0f617b2b074e1f2089b72b2a2eb2890
Author: popcornmix <popcornmix@gmail.com>
Date:   Mon Jun 11 17:29:55 2018 +0100

    configfs: hack: make it build

diff --git a/drivers/of/configfs.c b/drivers/of/configfs.c
index 178f0629b0f0..9f608b2569da 100644
--- a/drivers/of/configfs.c
+++ b/drivers/of/configfs.c
@@ -64,7 +64,7 @@ static int create_overlay(struct cfs_overlay_item *overlay, void *blob)
 	}
 	pr_debug("%s: resolved OK\n", __func__);
 
-	err = of_overlay_apply(overlay->overlay, &overlay->ov_id);
+	err = -1; // TODO: fix: of_overlay_apply(overlay->overlay, &overlay->ov_id);
 	if (err < 0) {
 		pr_err("%s: Failed to create overlay (err=%d)\n",
 				__func__, err);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1464_4a62ccc0e0f617b2b074e1f2089b72b2a2eb2890.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1464_4a62ccc0e0f617b2b074e1f2089b72b2a2eb2890.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
