commit 87b0390dc9942eca7f104ecaf0cdc57f31af9437
Author: Dan Pasanen <dan.pasanen@gmail.com>
Date:   Wed Sep 20 10:17:41 2017 -0500

    vcsm: fix multi-platform build

diff --git a/drivers/char/broadcom/vc_sm/vmcs_sm.c b/drivers/char/broadcom/vc_sm/vmcs_sm.c
index 034ae2f27f87..b7814d67500b 100644
--- a/drivers/char/broadcom/vc_sm/vmcs_sm.c
+++ b/drivers/char/broadcom/vc_sm/vmcs_sm.c
@@ -223,6 +223,7 @@ typedef void cache_flush_op_fn(const void *, const void *);
 #if defined(CONFIG_CPU_CACHE_V7)
 extern cache_flush_op_fn v7_dma_inv_range;
 extern cache_flush_op_fn v7_dma_clean_range;
+extern cache_flush_op_fn v7_dma_flush_range;
 static cache_flush_op_fn * const flushops[4] =
 {
 	0,
@@ -233,6 +234,7 @@ static cache_flush_op_fn * const flushops[4] =
 #elif defined(CONFIG_CPU_CACHE_V6)
 extern cache_flush_op_fn v6_dma_inv_range;
 extern cache_flush_op_fn v6_dma_clean_range;
+extern cache_flush_op_fn v6_dma_flush_range;
 static cache_flush_op_fn * const flushops[4] =
 {
 	0,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1132_87b0390dc9942eca7f104ecaf0cdc57f31af9437.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1132_87b0390dc9942eca7f104ecaf0cdc57f31af9437.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
