commit 8a697a50c090053cf6c53a096972dd841332fe59
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Thu Sep 7 08:35:53 2017 +0200

    Linux 4.9.48

diff --git a/Makefile b/Makefile
index a0abbfc15a49..cfa188b427b1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 9
-SUBLEVEL = 47
+SUBLEVEL = 48
 EXTRAVERSION =
 NAME = Roaring Lionus
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1707_8a697a50c090053cf6c53a096972dd841332fe59.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1707_8a697a50c090053cf6c53a096972dd841332fe59.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
