commit 1544c9368b75418d45fb6aebb1c61be745e5d8de
Author: Dan Pasanen <dan.pasanen@gmail.com>
Date:   Thu Sep 21 09:57:44 2017 -0500

    vcsm: add macros for cache functions

diff --git a/drivers/char/broadcom/vc_sm/vc_sm_knl.h b/drivers/char/broadcom/vc_sm/vc_sm_knl.h
index f7f74750d835..e50fd8eaecef 100644
--- a/drivers/char/broadcom/vc_sm/vc_sm_knl.h
+++ b/drivers/char/broadcom/vc_sm/vc_sm_knl.h
@@ -27,6 +27,11 @@ enum vc_sm_lock_cache_mode {
 	VC_SM_LOCK_NON_CACHED,
 };
 
+/* Cache functions */
+#define VCSM_CACHE_OP_INV       0x01
+#define VCSM_CACHE_OP_FLUSH     0x02
+#define VCSM_CACHE_OP_CLEAN     0x03
+
 /* Allocate a shared memory handle and block. */
 int vc_sm_alloc(struct vc_sm_alloc_t *alloc, int *handle);
 
diff --git a/drivers/char/broadcom/vc_sm/vmcs_sm.c b/drivers/char/broadcom/vc_sm/vmcs_sm.c
index 754f29e88968..80c5381a64c2 100644
--- a/drivers/char/broadcom/vc_sm/vmcs_sm.c
+++ b/drivers/char/broadcom/vc_sm/vmcs_sm.c
@@ -2964,12 +2964,9 @@ static long vc_sm_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 			}
 			for (i = 0; i < sizeof(ioparam.s) / sizeof(*ioparam.s); i++) {
 				switch (ioparam.s[i].cmd) {
-				default:
-				case 0:
-					break; /* NOOP */
-				case 1:	/* L1/L2 invalidate virtual range */
-				case 2: /* L1/L2 clean physical range */
-				case 3: /* L1/L2 clean+invalidate all */
+				case VCSM_CACHE_OP_INV:	/* L1/L2 invalidate virtual range */
+				case VCSM_CACHE_OP_FLUSH: /* L1/L2 clean physical range */
+				case VCSM_CACHE_OP_CLEAN: /* L1/L2 clean+invalidate all */
 					/* Locate resource from GUID. */
 					resource =
 					    vmcs_sm_acquire_resource(file_data, ioparam.s[i].handle);
@@ -2993,6 +2990,8 @@ static long vc_sm_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 						vmcs_sm_release_resource(resource, 0);
 
 					break;
+				default:
+					break; /* NOOP */
 				}
 			}
 		}
