commit 9c297d9ebd0d8c6d5e6e8675d527d6a221613abc
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Jun 30 14:12:42 2015 +0100

    serial: 8250: Don't crash when nr_uarts is 0

diff --git a/drivers/tty/serial/8250/8250_core.c b/drivers/tty/serial/8250/8250_core.c
index e8819aa20415..cf9c7d2e3f95 100644
--- a/drivers/tty/serial/8250/8250_core.c
+++ b/drivers/tty/serial/8250/8250_core.c
@@ -509,6 +509,8 @@ static void __init serial8250_isa_init_ports(void)
 
 	if (nr_uarts > UART_NR)
 		nr_uarts = UART_NR;
+	if (!nr_uarts)
+		return;
 
 	for (i = 0; i < nr_uarts; i++) {
 		struct uart_8250_port *up = &serial8250_ports[i];
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1009_9c297d9ebd0d8c6d5e6e8675d527d6a221613abc.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1009_9c297d9ebd0d8c6d5e6e8675d527d6a221613abc.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
