Description:  Fix usbip "format-overflow" failure.
Author: Peter Michael Green <plugwash@raspbian.org>


Index: linux-4.9-4.9.51/tools/usb/usbip/libsrc/usbip_common.c
===================================================================
--- linux-4.9-4.9.51.orig/tools/usb/usbip/libsrc/usbip_common.c
+++ linux-4.9-4.9.51/tools/usb/usbip/libsrc/usbip_common.c
@@ -214,10 +214,11 @@ int read_usb_device(struct udev_device *
 int read_usb_interface(struct usbip_usb_device *udev, int i,
 		       struct usbip_usb_interface *uinf)
 {
-	char busid[SYSFS_BUS_ID_SIZE];
+	char busid[SYSFS_BUS_ID_SIZE*2];
 	struct udev_device *sif;
 
-	sprintf(busid, "%s:%d.%d", udev->busid, udev->bConfigurationValue, i);
+	snprintf(busid,SYSFS_BUS_ID_SIZE*2, "%s:%d.%d", udev->busid, udev->bConfigurationValue, i);
+	busid[SYSFS_BUS_ID_SIZE-1] = 0;
 
 	sif = udev_device_new_from_subsystem_sysname(udev_context, "usb", busid);
 	if (!sif) {
Index: linux-4.9-4.9.51/tools/usb/usbip/libsrc/usbip_host_common.c
===================================================================
--- linux-4.9-4.9.51.orig/tools/usb/usbip/libsrc/usbip_host_common.c
+++ linux-4.9-4.9.51/tools/usb/usbip/libsrc/usbip_host_common.c
@@ -21,6 +21,9 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#pragma GCC diagnostic warning "-Wpragmas"
+#pragma GCC diagnostic warning "-Wformat-truncation"
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
Index: linux-4.9-4.9.51/tools/usb/usbip/src/usbip.c
===================================================================
--- linux-4.9-4.9.51.orig/tools/usb/usbip/src/usbip.c
+++ linux-4.9-4.9.51/tools/usb/usbip/src/usbip.c
@@ -176,6 +176,9 @@ int main(int argc, char *argv[])
 			break;
 		case '?':
 			printf("usbip: invalid option\n");
+			usbip_usage();
+			goto out;
+			break;
 		default:
 			usbip_usage();
 			goto out;
